shared/api/image_decoder_win32.hpp

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+#pragma once
+
+#pragma comment(lib, "Windowscodecs.lib")
+
+#include <wincodec.h>
+#include <wincodecsdk.h>
+
+#include "op_def_struct.h"
+#include "ext_status.h"
+
+
+inline OrtxStatus image_decoder(const ortc::Tensor<uint8_t>& input, ortc::Tensor<uint8_t>& output) {
+  const auto& dimensions = input.Shape();
+  if (dimensions.size() != 1ULL) {
+    return {kOrtxErrorInvalidArgument, "[ImageDecoder]: Only raw image formats are supported."};
+  }
+
+  // Get data & the length
+  const uint8_t* encoded_image_data = input.Data();
+  const int64_t encoded_image_data_len = input.NumberOfElement();
+
+  // check whether it's too small for a image
+  if (encoded_image_data_len < 8) {
+    return {kOrtxErrorInvalidArgument, "[ImageDecoder]: Invalid image data."};
+  }
+
+  OrtxStatus status{};
+
+  CoInitialize(NULL);
```

Comment:
In this case, we change this operator type to be a struct which has an Init method to avoid CoInitialize on every compute call, Here is the example: https://github.com/microsoft/onnxruntime-extensions/blob/8d842d85e39aa36985cedf68f5d9e5dfef6f6d05/shared/api/image_transforms.hpp#L35C1-L36C1

Reply:
sure